### PR TITLE
Feature/terra3 51 scheduled https api call

### PR DESCRIPTION
**Pull request by Kaushikkatariya**
Please review my PR.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -345,3 +345,62 @@ module "deployment_user" {
 
   source = "./modules/deployment_user"
 }
+
+# ---------------------------------------------------------------------------------------------------------------------
+# scheduled HTTPS API call
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "eventbridge" {
```
Comment (w13k) [inline main.tf]: Please move up everything into the scheduled_api_call module. It is there anyway and at a later stage we plan to allow 1 to many scheduled api calls.

File: main.tf [hunk 2]
```diff
@@ -345,3 +345,62 @@ module "deployment_user" {
...
+  source  = "terraform-aws-modules/lambda/aws"
+  version = "~> 2.0"
+
+  function_name = "terra3-https-api-call"
+  description   = "scheduled https api call"
+  handler       = "index.handler"
+  runtime       = "nodejs16.x"
+
+  create_package         = false
+  local_existing_package = data.archive_file.function[0].output_path
```
Comment (w13k) [inline main.tf]: Don't create the zip by yourself with data.archive_file but rather use what's already given by the lambda module, namely the "source_path". Move up the index.js as well into the scheduled_api_call folder and rename it to something more self-explaining, e.g. api_call.js

File: variables.tf [hunk 3]
```diff
@@ -3,7 +3,7 @@ variable "solution_name" {
   type        = string
 
   validation {
-    condition     = length(var.solution_name) <= 16 && can(regex("^([a-z0-9])+(?:-[a-z0-9]+)*$", var.solution_name))
+    condition     = length(var.solution_name) <= 26 && can(regex("^([a-z0-9])+(?:-[a-z0-9]+)*$", var.solution_name))
     error_message = "Only max. 16 lower-case alphanumeric characters and dashes in between are allowed."
```
Comment (w13k) [inline variables.tf]: 26 or 16? Currently, the condition and the description do not match.

File: examples/scheduled_api_call/main.tf [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# ---------------------------------------------------------------------------------------------------------------------
```
Comment (w13k) [inline examples/scheduled_api_call/main.tf:1]: Before working on the feedback, please pull the branch as I've pushed already some slights refactorings by myself!

**Review by w13k — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (Kaushikkatariya) [reply main.tf]: Done

File: main.tf (hunk 2 above)
Comment (Kaushikkatariya) [reply main.tf]: done

File: main.tf (hunk 1 above)
Comment (Kaushikkatariya) [reply main.tf]: I have disabled module "scheduled_api_call" in modules/scheduled_api_call, as it requires the api_key.

**Review by w13k — APPROVED** [review]

**Review by w13k — COMMENTED** [review]
When running the example, I get this error message. Can you please have a look?

image

**Comment by w13k** [comment]
This PR is included in version 0.12.0 :tada:

**Pull request merged**